tests/bdd/features/rebuild/test_log_based_rebuild.py

```diff
@@ -0,0 +1,297 @@
+
+
+@when("the volume target moves to a different node before the log-based rebuild starts")
+def _():
+    """the volume target moves to a different node before the log-based rebuild starts."""
+    raise NotImplementedError
+
+
+@then("a full rebuild starts after some time")
+def _():
+    """a full rebuild starts after some time."""
+    nexus = ApiClient.nexuses_api().get_node_nexuses(NODE_1_NAME)
+    assert len(nexus[0].children) == NUM_VOLUME_REPLICAS
+    childlist = nexus[0].children
+    for child in childlist:
+        if child.state == ChildState("Faulted") or child.state == ChildState(
+            "Degraded"
+        ):
+            assert child.uri != faulted_child_uri
+            assert child.rebuild_progress != 0
+
+
+@then("a log-based rebuild is started after reconnection")
+def _():
+    """a log-based rebuild is started after reconnection."""
+    raise NotImplementedError
+
+
+@then("a log-based rebuild is started after the io-engine restarts")
+def _():
+    """a log-based rebuild is started after the io-engine restarts."""
+    raise NotImplementedError
+
+
+@then("a log-based rebuild starts after some time")
+def _():
+    """a log-based rebuild starts after some time."""
+    time.sleep(0.5)
+    log_based_rebuild_is_running()
+
+
+@then("log-based rebuild is running")
+def log_based_rebuild_is_running():
+    """log-based rebuild is running."""
+    # TODO: Check log-based rebuild is running.
+    # No direct way to check partial rebuild via openapi yet
+    nexus = ApiClient.nexuses_api().get_node_nexuses(NODE_1_NAME)
+    assert len(nexus[0].children) == NUM_VOLUME_REPLICAS
+    childlist = nexus[0].children
+    for child in childlist:
+        if child.state == ChildState("Faulted") or child.state == ChildState(
+            "Degraded"
+        ):
+            assert child.uri == faulted_child_uri
+            assert child.rebuild_progress != 0
+
+
+@then("the node hosting rebuilding replica crashes permanently")
+def _():
+    """the node hosting rebuilding replica crashes permanently."""
+    raise NotImplementedError
+
+
+@then("the replica is not online again within the timeout period")
+def _():
+    """the replica is not online again within the timeout period."""
+    raise NotImplementedError
+
+
+@then("the volume target moves to a different node before rebuild completion")
+def _():
+    """the volume target moves to a different node before rebuild completion."""
+    raise NotImplementedError
+
+
+@retry(wait_fixed=1000, stop_max_attempt_number=10)
+def wait_for_degraded_volume():
+    volume = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    assert volume.state.status == VolumeStatus("Degraded")
+
+
+@retry(wait_fixed=1000, stop_max_attempt_number=10)
+def wait_for_pool_online(pool_uuid):
+    pool = ApiClient.pools_api().get_pool(pool_uuid)
+    assert pool.state.status == PoolStatus("Online")
+
+
+def remove_stale_files_on_error(ip, nqn):
```

Comment:
fio should take care of this by itself now IIRC

Reply:
will check this once fio usage in .py file is fixed. Have kept fio run commented for now.